### PR TITLE
feat(component): add description to select option and select action

### PR DESCRIPTION
**Pull request by golcinho**
## What?
Adding a description to the `select option` and `select action`.

## Screenshots
Screenshot 2020-07-28 at 11 15 05
Screenshot 2020-07-30 at 09 45 23
Screenshot 2020-07-30 at 09 45 38
Screenshot 2020-07-28 at 09 38 44
Screenshot 2020-07-28 at 09 39 01

## Testing
Unit tests.

### CODE REVIEW

**Comment by valterfatia** [comment]
Nice work! Two small things from me:
1. The description text should be in "secondary60" and "small text" – or was there a different direction from another designer?
2. I'm unsure about changing the colour of the description when there's an interaction – currently reviewing within the design team, hoping to have it decided soon ;)

**Comment by golcinho** [comment]
> Nice work! Two small things from me:
> 
> 1. The description text should be in "secondary60" and "small text" – or was there a different direction from another designer?
> 2. I'm unsure about changing the colour of the description when there's an interaction – currently reviewing within the design team, hoping to have it decided soon ;)

Hey @valterfatia !

Thank you for the comments!

I changed color to `secondary60` and description text is `Small` (14px).

Thank you for your assistance with the second point 🤝

**Comment by golcinho** [comment]
> Everything looks good, but do you think we can add these changes to the `MultiSelect` component as well?
> 
> Thanks for adding this! 👍

Good point, thank you 🙇 

I will make it for `MultiSelect` too 👍

File: packages/docs/pages/Select/SelectPage.tsx [hunk 1]
```diff
@@ -342,6 +342,53 @@ const SelectPage = () => (
       </Form>
       {/* jsx-to-string:end */}
     </CodePreview>
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and select action</Text>
```
Comment (chanceaclark) [inline packages/docs/pages/Select/SelectPage.tsx:348]:
```suggestion
    <Text>It is possible to add a description for select options and actions</Text>
```

File: packages/big-design/src/components/Select/Select.tsx [hunk 2]
```diff
@@ -465,3 +480,15 @@ const iconColor = <T extends any>(item: SelectOption<T> | SelectAction, isHighli
...
+const descriptionColor = <T extends any>(item: SelectOption<T> | SelectAction, isHighlighted: boolean) => {
+  if (item.disabled) {
+    return 'secondary40';
+  }
+
+  if (!isHighlighted || !('onActionClick' in item)) {
+    return 'secondary60';
+  }
+
+  return 'actionType' in item ? (item.actionType === 'destructive' ? 'danger30' : 'primary30') : 'primary30';
```
Comment (chanceaclark) [inline packages/big-design/src/components/Select/Select.tsx:493]: @valterfatia `danger30` color with `danger10` background-color doesn't meet accessibility color contrast ratios. Is there another option?

**Comment by valterfatia** [comment]
Clarifying the last point, from my previous comment lets keep the text as secondary60 on hover and just do the background change as we already have.

File: packages/big-design/src/components/Select/Select.tsx (hunk 2 above)
Comment (valterfatia) [reply packages/big-design/src/components/Select/Select.tsx:493]: I have been reviewing that change within the design team and we have agreed to keep secondary60 as the text colour to meet accessibility requirements.

**Comment by valterfatia** [comment]
Okay I have one more request – which is likely better addressed in a separate PR/issue – but I wanted to call it out here.

All text in a dropdown should remain left aligned to a single "key line" regardless of the presence of icons within the dropdown. In practice this when an item in a dropdown has an icon, then other elements without an icon should keep the same gap so the text aligns in a single line.

Ideally we should aim for having dropdowns items should all have a consistent look and feel, but we also accept in some instances this won't be possible. In those cases our request is for the space to be left empty.

**Comment by chanceaclark** [comment]
> Okay I have one more request – which is likely better addressed in a separate PR/issue – but I wanted to call it out here.
> 
> All text in a dropdown should remain left aligned to a single "key line" regardless of the presence of icons within the dropdown. In practice this when an item in a dropdown has an icon, then other elements without an icon should keep the same gap so the text aligns in a single line.
> 
> Ideally we should aim for having dropdowns items should all have a consistent look and feel, but we also accept in some instances this won't be possible. In those cases our request is for the space to be left empty.

Let's talk about this offline in our Feedback Session.

**Comment by valterfatia** [comment]
> Let's talk about this offline in our Feedback Session.

Sounds good!

**Comment by golcinho** [comment]
@chanceaclark @valterfatia I changed description color to `secondary60` for the select options and actions. (screenshots are updated).

Also, I added a description of the options and actions for the `MultiSelect` component.

cc @jorgemoya

File: packages/docs/pages/MultiSelect/MultiSelectPage.tsx [hunk 3]
```diff
@@ -300,6 +300,53 @@ const MultiSelectPage = () => (
       </Form>
       {/* jsx-to-string:end */}
     </CodePreview>
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and select action</Text>
```
Comment (chanceaclark) [inline packages/docs/pages/MultiSelect/MultiSelectPage.tsx:306]:
```suggestion
    <Text>It is possible to add a description for select options and actions.</Text>
```

File: packages/docs/pages/Select/SelectPage.tsx [hunk 4]
```diff
@@ -342,6 +342,53 @@ const SelectPage = () => (
       </Form>
       {/* jsx-to-string:end */}
     </CodePreview>
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and actions</Text>
```
Comment (chanceaclark) [inline packages/docs/pages/Select/SelectPage.tsx:348]:
```suggestion
    <Text>It is possible to add a description for select options and actions.</Text>
```

**Pull request merged**